contracts/multisigs/GnosisSafeSameAddressMultisig.sol

```diff
@@ -31,41 +31,70 @@ error WrongNumOwners(uint256 expected, uint256 provided);
 /// @param provided Provided owner address.
 error WrongOwner(address provided);
 
+/// @dev Multisig transaction resulted in a failure.
+/// @param provided Provided multisig address.
+error MultisigExecFailed(address provided);
+
 /// @title Gnosis Safe Same Address - Smart contract for Gnosis Safe verification of an already existent multisig address.
 /// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
 contract GnosisSafeSameAddressMultisig {
     // Default data size to be parsed as an address of a Gnosis Safe multisig proxy address
+    // This exact size suggests that all the changes to the multisig have been performed and only validation is needed
     uint256 public constant DEFAULT_DATA_LENGTH = 20;
 
-    /// @dev Verifies the existent gnosis safe multisig for changed owners and threshold.
-    /// @notice The multisig is supposedly updated before reaching this step such that new multisig is not created.
-    ///         The order of owners' addresses in the multisig is in reverse order compared to how they were added:
-    ///         https://etherscan.io/address/0xd9db270c1b5e3bd161e8c8503c55ceabee709552#code#F6#L56
-    /// @param owners Set of multisig owners.
-    /// @param threshold Number of required confirmations for a multisig transaction.
-    /// @param data Packed data containing address of an existent gnosis safe multisig.
+    /// @dev Updates and/or verifies the existent gnosis safe multisig for changed owners and threshold.
+    /// @notice This function operates with existent multisig proxy that is requested to be updated in terms of
+    ///         the set of owners' addresses and the threshold. There are two scenarios possible:
+    ///         1. The multisig proxy is already updated before reaching this function. Then the multisig address
+    ///            must be passed as a payload such that its owners and threshold are verified against those specified
+    ///            in the argument list.
+    ///         2. The multisig proxy is not yet updated. Then the multisig address must be passed in a packed bytes of
+    ///            data along with the Gnosis Safe `execTransaction()` function arguments packed payload. That payload
+    ///            is going to modify the mulsisig proxy as per its signed transaction. At the end, the updated multisig
+    ///            proxy is going to be verified with the provided set of owners' addresses and the threshold.
+    ///         Note that the order of owners' addresses in the multisig is in reverse order, see comments below.
```

Comment:
Please remove “see comments below” and replace with full explanation (so natspec is standalone).